packages/EasyDecision/src/Interfaces/RestrictedRuleInterface.php

```diff
@@ -0,0 +1,10 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EonX\EasyDecision\Interfaces;
+
+interface RestrictedRuleInterface extends RuleInterface
```

Comment:
Maybe we should create some default abstract restricted rule, because the most common use case to restrict a rule I can think of would be based on its name. So maybe an abstract "name restricted" rule that implements `supportsDecision()` and automatically check the decision name against the one define by the rule, using an abstract protected method, could be cool